### ISSUE TITLE
K780 unable to change fn-swap settings from cli on 1.0.3

### ISSUE THREAD

**Issue by makarovks** [pwr-Solaar/Solaar#941]
**Information**

- Solaar version (`solaar --version` or `git describe --tags` if cloned from this repository): 1.0.3
- Distribution: Manjaro
- Kernel version (ex. `uname -srmo`): Linux 5.4.60-2-MANJARO x86_64 GNU/Linux
- Output of `solaar show`:
```
Unifying Receiver
  Device path  : /dev/hidraw0
  USB id       : 046d:c52b
  Serial       : EEB3971F
    Firmware   : 12.09.B0030
    Bootloader : 04.16
    Other      : AA.AA
  Has 2 paired device(s) out of a maximum of 6.
  Notifications: wireless, software present (0x000900)
  Device activity counters: 1=183, 2=198

  1: Wireless Multi-Device Keyboard K780
     Codename     : K780
     Kind         : keyboard
     Wireless PID : 405B
     Protocol     : HID++ 4.5
     Polling rate : 20 ms (50Hz)
     Serial number: 2851CF4A
        Bootloader: BOT 25.01.B0006
          Firmware: MPK 01.05.B0029
             Other: 
     The power switch is located on the edge of top right corner.
     Supports 31 HID++ 2.0 features:
         0: ROOT                   {0000}   
         1: FEATURE SET            {0001}   
         2: DEVICE FW VERSION      {0003}   
         3: DEVICE NAME            {0005}   
         4: WIRELESS DEVICE STATUS {1D4B}   
         5: RESET                  {0020}   
         6: DEVICE FRIENDLY NAME   {0007}   
         7: BATTERY STATUS         {1000}   
         8: CHANGE HOST            {1814}   
            Change Host: 1:kirill-brix
         9: HOSTS INFO             {1815}   
            Host 0 (paired): 
            Host 1 (paired): MEIZU M5 Note
            Host 2 (unpaired): 
        10: REPROG CONTROLS V4     {1B04}   
        11: PERSISTENT REMAPPABLE ACTION {1C00}   
        12: NEW FN INVERSION       {40A2}   
            Fn-swap: disabled
            Fn-swap default: enabled
            Swap Fx function: False
        13: ENCRYPTION             {4100}   
        14: KEYBOARD DISABLE KEYS  {4521}   
            Disable keys: {'1': False, '2': False, '4': False, '8': False, '16': False}
        15: MULTIPLATFORM          {4531}   
            Set OS: Windows
        16: LOCK KEY STATE         {4220}   
        17: DFUCONTROL SIGNED      {00C2}   
        18: unknown:1803           {1803}   internal, hidden
        19: CONFIG DEVICE PROPS    {1806}   internal, hidden
        20: OOBSTATE               {1805}   internal, hidden
        21: unknown:1813           {1813}   internal, hidden
        22: unknown:1830           {1830}   internal, hidden
        23: unknown:1861           {1861}   internal, hidden
        24: unknown:1890           {1890}   internal, hidden
        25: unknown:1891           {1891}   internal, hidden
        26: unknown:18A1           {18A1}   internal, hidden
        27: unknown:1DF3           {1DF3}   internal, hidden
        28: unknown:1E00           {1E00}   hidden
        29: unknown:1EB0           {1EB0}   internal, hidden
        30: unknown:18B0           {18B0}   internal, hidden
     Has 19 reprogrammable keys:
         0: Host Switch channel 1     , default: HostSwitch Channel 1        => Host Switch channel 1     
             is FN, FN sensitive, divertable, persistently divertable, pos:1, group:0, gmask:0
         1: Host Switch channel 2     , default: HostSwitch Channel 2        => Host Switch channel 2     
             is FN, FN sensitive, divertable, persistently divertable, pos:2, group:0, gmask:0
         2: Host Switch channel 3     , default: HostSwitch Channel 3        => Host Switch channel 3     
             is FN, FN sensitive, divertable, persistently divertable, pos:3, group:0, gmask:0
         3: Multiplatform Home/Mission Control, default: Multiplatform Home/Mission Control => Multiplatform Home/Mission Control
             is FN, FN sensitive, reprogrammable, divertable, persistently divertable, pos:4, group:0, gmask:0
         4: Multiplatform Menu/Show/Hide Virtual Keyboard/Launchpad, default: Multiplatform Menu/Launchpad => Multiplatform Menu/Show/Hide Virtual Keyboard/Launchpad
             is FN, FN sensitive, reprogrammable, divertable, persistently divertable, pos:5, group:0, gmask:0
         5: MultiPlatform Back        , default: MultiPlatform Back          => MultiPlatform Back        
             is FN, FN sensitive, reprogrammable, divertable, persistently divertable, pos:6, group:0, gmask:0
         6: Previous                  , default: Previous                    => Previous                  
             is FN, FN sensitive, divertable, persistently divertable, pos:7, group:0, gmask:0
         7: Play/Pause                , default: Play/Pause                  => Play/Pause                
             is FN, FN sensitive, divertable, persistently divertable, pos:8, group:0, gmask:0
         8: Next                      , default: Next                        => Next                      
             is FN, FN sensitive, divertable, persistently divertable, pos:9, group:0, gmask:0
         9: Mute                      , default: Mute                        => Mute                      
             is FN, FN sensitive, divertable, persistently divertable, pos:10, group:0, gmask:0
        10: Volume Down               , default: Volume Down                 => Volume Down               
             is FN, FN sensitive, divertable, persistently divertable, pos:11, group:0, gmask:0
        11: Volume Up                 , default: Volume Up                   => Volume Up                 
             is FN, FN sensitive, divertable, persistently divertable, pos:12, group:0, gmask:0
        12: Multiplatform search      , default: Multiplatform Search        => Multiplatform search      
             nonstandard, reprogrammable, divertable, persistently divertable, pos:0, group:0, gmask:0
        13: MultiPlatform Insert      , default: Switch Language             => MultiPlatform Insert      
             is FN, reprogrammable, divertable, persistently divertable, pos:0, group:0, gmask:0
        14: Screen Capture/Print Screen, default: Screen Capture              => Screen Capture/Print Screen
             is FN, reprogrammable, divertable, persistently divertable, pos:0, group:0, gmask:0
        15: Multiplatform Lock        , default: WindowsLock                 => Multiplatform Lock        
             is FN, reprogrammable, divertable, persistently divertable, pos:0, group:0, gmask:0
        16: Fn Up                     , default: unknown:0070                => Fn Up                     
             is FN, divertable, persistently divertable, pos:0, group:0, gmask:0
        17: Fn Down                   , default: unknown:006F                => Fn Down                   
             is FN, divertable, persistently divertable, pos:0, group:0, gmask:0
        18: unknown:0034              , default: Do Nothing One              => unknown:0034              
             nonstandard, pos:0, group:0, gmask:0
     Battery: 20%, discharging, next level 5%.

  2: Wireless Mouse M335
     Codename     : M335
     Kind         : mouse
     Wireless PID : 4050
     Protocol     : HID++ 4.5
     Polling rate : 8 ms (125Hz)
     Serial number: 13BC865D
          Firmware: RQM 58.01.B0010
     The power switch is located on the base.
     Supports 22 HID++ 2.0 features:
         0: ROOT                   {0000}   
         1: FEATURE SET            {0001}   
         2: DEVICE FW VERSION      {0003}   
         3: DEVICE NAME            {0005}   
         4: RESET                  {0020}   
         5: BATTERY STATUS         {1000}   
         6: DEVICE RESET           {1802}   internal, hidden
         7: unknown:1810           {1810}   internal, hidden
         8: unknown:1830           {1830}   internal, hidden
         9: unknown:1862           {1862}   internal, hidden
        10: unknown:1890           {1890}   internal, hidden
        11: unknown:18A0           {18A0}   internal, hidden
        12: unknown:18B1           {18B1}   internal, hidden
        13: REPROG CONTROLS V4     {1B04}   
            Actions: {'80': 80, '81': 81, '82': 82, '83': 83, '86': 86, '204': 204}
        14: WIRELESS DEVICE STATUS {1D4B}   
        15: REMAINING PAIRING      {1DF0}   hidden
        16: unknown:1DF3           {1DF3}   internal, hidden
        17: unknown:1E00           {1E00}   hidden
        18: unknown:1EB0           {1EB0}   internal, hidden
        19: unknown:1F03           {1F03}   internal, hidden
        20: HIRES WHEEL            {2121}   
            Multiplier: 1
            Has invert
              Normal wheel motion
            High resolution mode
            HID notification
            High Resolution Wheel Invert: False
            Wheel Resolution: True
        21: POINTER SPEED          {2205}   
            Pointer Speed: 1.0
            Sensitivity (Pointer Speed): 256
     Has 6 reprogrammable keys:
         0: Left Button               , default: Left Click                  => Left Button               
             mse, divertable, pos:0, group:1, gmask:1
         1: Right Button              , default: Right Click                 => Right Button              
             mse, divertable, pos:0, group:1, gmask:1
         2: Middle Button             , default: Mouse Middle Button         => Middle Button             
             mse, reprogrammable, divertable, pos:0, group:2, gmask:3
         3: Back Button               , default: Mouse Back Button           => Back Button               
             mse, reprogrammable, divertable, pos:0, group:2, gmask:3
         4: Forward Button            , default: Mouse Forward Button        => Forward Button            
             mse, reprogrammable, divertable, pos:0, group:2, gmask:3
         5: Switch App                , default: AppSwitch                   => Switch App                
             mse, reprogrammable, divertable, pos:0, group:2, gmask:3
     Battery: 70%, discharging, next level 5%.
```

**Describe the bug**
When trying to change setting of fn-swap of K780 keyboard from cli:
```
>solaar config K780 fn-swap false
solaar: error: Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/solaar/cli/__init__.py", line 175, in run
    m.run(c, args, _find_receiver, _find_device)
  File "/usr/lib/python3.8/site-packages/solaar/cli/config.py", line 74, in run
    raise Exception("no setting '%s' for %s" % (args.setting, dev.name))
Exception: no setting 'fn-swap' for Wireless Multi-Device Keyboard K780
```
But if list all settings of K780 we can see this setting in list.
```
>solaar config K780
Wireless Multi-Device Keyboard K780 (K780) [405B:2851CF4A]

# Swap Fx function
# When set, the F1..F12 keys will activate their special function, and you must hold the FN key to activate their standard function.  When unset, the F1..F12 keys will activate their standard function, and you must hold the FN key to activate their special function.
#   possible values: on/true/t/yes/y/1 or off/false/f/no/n/0
fn-swap = False

# Disable keys
# Disable specific keyboard keys.
disable-keyboard-keys = {'1': False, '2': False, '4': False, '8': False, '16': False}

# Set OS
# Change keys to match OS.
#   possible values: one of [ Windows, MacOS, iOS ], or higher/lower/highest/max/lowest/min
multiplatform = NamedInt(0, 'Windows')

# Change Host
# Switch connection to a different host
#   possible values: one of [ 1:kirill-brix, 2:MEIZU M5 Note, 3 ], or higher/lower/highest/max/lowest/min
change-host = NamedInt(0, '1:kirill-brix')
```
From UI all working just fine but in debug mode have an errors:
```
14:36:49,598     INFO [AsyncUI] solaar.configuration: saved {'0640:13BC865D': {'_name': 'M335', 'hires-smooth-invert': False, 'hires-smooth-resolution': True, 'pointer_speed': 256}, '1A40:2851CF4A': {'_name': 'Wireless Multi-Device Keyboard K780', 'fn-swap': False}, '4050:13BC865D': {'_name': 'Wireless Mouse M335', 'hires-smooth-invert': False, 'hires-smooth-resolution': True, 'pointer_speed': 256, 'reprogrammable-keys': {'204': 204, '80': 80, '81': 81, '82': 82, '83': 83, '86': 86}}, '4050:EEB3971F': {'_name': 'M335', 'hires-smooth-invert': False, 'hires-smooth-resolution': True, 'pointer_speed': 256}, '405B:2851CF4A': {'_name': 'Wireless Multi-Device Keyboard K780', 'disable-keyboard-keys': {'1': False, '16': False, '2': False, '4': False, '8': False}, 'fn-swap': False, 'multiplatform': 0}, '405B:EEB3971F': {'_name': 'Wireless Multi-Device Keyboard K780', 'fn-swap': False}, '_version': '1.0.3'} to /home/kmakarov/.config/solaar/config.json
14:36:57,819     INFO [AsyncUI] solaar.configuration: saved {'0640:13BC865D': {'_name': 'M335', 'hires-smooth-invert': False, 'hires-smooth-resolution': True, 'pointer_speed': 256}, '1A40:2851CF4A': {'_name': 'Wireless Multi-Device Keyboard K780', 'fn-swap': False}, '4050:13BC865D': {'_name': 'Wireless Mouse M335', 'hires-smooth-invert': False, 'hires-smooth-resolution': True, 'pointer_speed': 256, 'reprogrammable-keys': {'204': 204, '80': 80, '81': 81, '82': 82, '83': 83, '86': 86}}, '4050:EEB3971F': {'_name': 'M335', 'hires-smooth-invert': False, 'hires-smooth-resolution': True, 'pointer_speed': 256}, '405B:2851CF4A': {'_name': 'Wireless Multi-Device Keyboard K780', 'disable-keyboard-keys': {'1': False, '16': False, '2': False, '4': False, '8': False}, 'fn-swap': True, 'multiplatform': 0}, '405B:EEB3971F': {'_name': 'Wireless Multi-Device Keyboard K780', 'fn-swap': False}, '_version': '1.0.3'} to /home/kmakarov/.config/solaar/config.json
14:36:59,686  WARNING [ReceiverListener:hidraw0] logitech_receiver.notifications: <PairedDevice(1,405B,K780,2851CF4A)>: unrecognized Notification(1,0A,00,00340000000000000000000000000000) for feature REPROG CONTROLS V4 (index 0A)
14:37:00,996  WARNING [ReceiverListener:hidraw0] logitech_receiver.notifications: <PairedDevice(1,405B,K780,2851CF4A)>: unrecognized Notification(1,0A,00,00000000000000000000000000000000) for feature REPROG CONTROLS V4 (index 0A)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/site-packages/logitech_receiver/listener.py", line 193, in run
    self._notifications_callback(n)
  File "/usr/lib/python3.8/site-packages/solaar/listener.py", line 209, in _notifications_handler
    _log.info('ignoring DJ pairing notification', n)
Message: 'ignoring DJ pairing notification'
Arguments: (_HIDPP_Notification(devnumber=1, sub_id=65, address=1, data=b'[@\x1a@\x00\x00\x00\x00\x00\x00\x00'),)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1081, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 925, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 664, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/site-packages/logitech_receiver/listener.py", line 193, in run
    self._notifications_callback(n)
  File "/usr/lib/python3.8/site-packages/solaar/listener.py", line 209, in _notifications_handler
    _log.info('ignoring DJ pairing notification', n)
Message: 'ignoring DJ pairing notification'
Arguments: (_HIDPP_Notification(devnumber=2, sub_id=65, address=1, data=b'P@\x06\x00\x00\x00\x00\x00\x00\x00\x00'),)
14:37:01,316  WARNING [ReceiverListener:hidraw0] solaar.listener: Unexpected device number (0) in notification Notification(0,41,02,0000000000000000000000).

```

**To Reproduce**
attach K780 keyboard
try to type command in console "solaar config K780 fn-swap false"

**Screenshots**

**Additional context**

**Comment by pfps** [comment]
The error you are seeing in debug mode has been fixed in a commit after release 1.0.3.  If you clone Solaar from this repository and run it you should not experience this problem.

The problem with config should also have been fixed in a commit after release 1.0.3.  Please try with the current version of Solaar from this repository.  If you still experience the problem, try with flag `-dd` and paste the output into this issue.

There was a bug having to do with the fact that there are multiple device features for the setting fn-swap and only the first one was being checked.  This bug was fixed after release 1.0.3.  I think that this bug is causing your problem.

**Comment by makarovks** [comment]
Yes, after update from git all is fine
Thank you!
